src/main/java/ch/uzh/ifi/ase/csccrecommender/mining/RecommendingLineContextVisitor.java

```diff
@@ -56,23 +57,33 @@ protected void handleCompletionExpression(String invocationType, CsccContext csc
   private void rankRecommendations(List<DocumentComparison> comparisons, String overallContext, String lineContext) {
     recommendations.clear();
     // TODO: Include Hamming distance from line context in sorting logic (unclear from paper description).
-    comparisons.sort(Comparator.comparingLong(DocumentComparison::getOverallContextHammingDistance));
+    // TODO: Done, check implementation in DocumentComparison.java file
 
+
+    //TODO: multiple thread sorting to improve performance (both first sort and second sort)
```

Comment:
Did you measure how long it takes now? If it only takes a few ms anyway it might not really be worth the effort.

Reply:
Not yet, I think it will matter when the index is big. I will test that later after I have finished the testcases.